packages/js/product-editor/src/hooks/use-product-template/use-product-template.ts

```diff
@@ -16,13 +16,27 @@ declare global {
 	}
 }
 
+const matchesAllTemplateMetaFields = (
+	templateMeta: Metadata< string >[],
+	productMeta: Metadata< string >[]
+) =>
+	templateMeta.every( ( item ) =>
+		productMeta.find(
+			( productMetaEntry ) =>
+				productMetaEntry.key === item.key &&
+				productMetaEntry.value === item.value
+		)
+	);
+
 export const useProductTemplate = (
 	productTemplateId: string | undefined,
-	productType: ProductType | undefined
+	product: Partial< Product >
```

Comment:
Do you expect/want `useProductTemplate` to be able to be called without a `product`? The code below checks whether product is not-nullish. If so, should update the signature to allow for `product` to not be passed in.